aspnetcore/aspnetcore-2.1.md

```diff
@@ -0,0 +1,188 @@
+
+### Configuration for production
+
+In production, HTTPS must be explicitly configured. In 2.1, default configuration schema for configuring HTTPS for Kestrel has been added. Applications can be configured to use:
+
+* Multiple endpoints including the URLs.
+* The certificate to use for HTTPS either from a file on disk or from a certificate store.
+
+## GDPR
+
+ASP.NET Core provides APIs and templates to help meet some of the [UE General Data Protection Regulation (GDPR)](https://www.eugdpr.org/) requirements. See [GDPR support in ASP.NET Core](xref:security/gdpr) for more information.  A [sample app](https://github.com/aspnet/Docs/tree/live/aspnetcore/security/gdpr/sample) lets you test most of the GDPR extension points and APIs added to the ASP.NET Core 2.1 templates.
+
+## Integration tests
+
+A new package is introduced that streamlines test creation and execution. The [Microsoft.AspNetCore.Mvc.Testing](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Testing/) package handles the following tasks:
+
+* Copies the dependencies file (*\*.deps*) from the tested app into the test project's *bin* folder.
+* Sets the content root to the tested app's project root so that static files and pages/views are found when the tests are executed.
+* Provides the [WebApplicationFactory](/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1) class to streamline bootstrapping the tested app with [TestServer](/dotnet/api/microsoft.aspnetcore.testhost.testserver).
+
+The following test uses [xUnit](https://xunit.github.io/) to check that the Index page loads with a success status code and with the correct Content-Type header:
+
+```csharp
+public class BasicTests 
+    : IClassFixture<WebApplicationFactory<RazorPagesProject.Startup>>
+{
+    private readonly HttpClient _client;
+        
+    public BasicTests(WebApplicationFactory<RazorPagesProject.Startup> factory)
+    {
+        _client = factory.CreateClient();
+    }
+
+    [Fact]
+    public async Task GetHomePage()
+    {
+        // Act
+        var response = await _client.GetAsync("/");
+
+        // Assert
+        response.EnsureSuccessStatusCode(); // Status Code 200-299
+        Assert.Equal("text/html; charset=utf-8", 
+            response.Content.Headers.ContentType.ToString());
+    }
+}
+```
+
+For more information, see the [Integration tests](xref:test/integration-tests) topic.
+
+## [ApiController], ActionResult
+
+In ASP.NET Core 2.1, improved support for OpenAPI specification(previously known as **Swagger**)) has been added to Web API. `ActionResult<T>` is a new type added to allow an application to return either a response type or any other action result (similar to IActionResult), while still indicating the response type. The `[ApiController] attribute has also been added as the way to opt-in to Web API-specific conventions and behaviors.
```

Comment:
~~In ASP.NET Core 2.1, improved support for OpenAPI specification(previously known as **Swagger**)) has been added to Web API.~~

ASP.NET Core 2.1 add new programming conventions that make it easier to build clean and descriptive web APIs. `AcitonResult<T>` . . .